pkg/device/device.go

```diff
@@ -124,85 +128,110 @@ func (x ListResponseType) String() string {
 	return proto.EnumName(ListresponsetypeName, int32(x))
 }
 
-func setAttribute(o *topo.Object, k string, v string) {
-	if len(v) > 0 {
-		o.Attributes[k] = v
-	}
-}
-
-func flag(b bool) string {
-	if b {
-		return "true"
-	}
-	return "false"
-}
-
-// ToObject converts topology object entity to local device
+// ToObject converts local device to a topology object entity.
 func ToObject(device *Device) *topo.Object {
-	o := &topo.Object{
-		ID:         topo.ID(device.ID),
-		Revision:   device.Revision,
-		Attributes: device.Attributes,
-		Type:       topo.Object_ENTITY,
-		Obj: &topo.Object_Entity{
-			Entity: &topo.Entity{
-				KindID:    topo.ID(device.Type),
-				Protocols: device.Protocols,
+	o := device.Object
+	if o == nil {
```

Comment:
Why this condition is needed?

Reply:
This is in case the original device object were synthesized from scratch.